pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -1269,13 +1268,7 @@ func encodeNetworkPortConfig(ctx *zedagentContext,
 	// XXX  string dhcpRangeHigh = 18;
 
 	dp.Proxy = encodeProxyStatus(&npc.ProxyConfig)
-	if !npc.ParseErrorTime.IsZero() {
-		errInfo := new(info.ErrorInfo)
-		errInfo.Description = npc.ParseError
-		errTime, _ := ptypes.TimestampProto(npc.ParseErrorTime)
-		errInfo.Timestamp = errTime
-		dp.Err = errInfo
-	}
+	dp.Err = encodeErrorInfo(npc.Error, npc.ErrorTime)
```

Comment:
Why did you remove the IsZero check?

Reply:
I moved it to encodeErrorInfo:

	if errStr == "" {
		return nil
	}

Changed it to errStr rather than timestamp..

Reply:
Why? Was there a bug? Why change it for one of of N cases?? 
Can we please focus on the thing which needs to be addressed? Which has nothing to do with syntactic changes around errors, but about reporting errors which are currently not.

Reply:
Ok.. I changed ParseError / ParseErrorTime to Error / ErrorTime since having two different types of errors were clearly not needed in the same struct.. So I had to touch this part..

And then, when I am encoding these errors - I did not like the repeated code and made a routine and fixed the places.
And when I am doing that - I noticed we are checking the timestamp rather than the error itself.. I felt checking the error is more appropriate.. And made the change.. 

I guess the point of discussion is - is checking errStr != nil better or timestamp != 0?